wallet/src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -65,11 +67,14 @@ newPayment
     -> Coin
     -> m CTx
 newPayment sa passphrase srcAccount dstAccount coin =
+    notFasterThan (1 :: Second) $  -- in order not to overflow relay
```

Comment:
@Martoon-00 Probably we are under time constraint but I think a better solution would be to capture the time before starting the action and sampling it at the end, and if the action took less than `time`, then wait for the remain of the time. This is because if you have a payment action which takes, say, 1.5 sec already, you are now waiting 1 second more for no good reason. Conversely, in case you have a payment taking 0.5 seconds you can sample the time the action ended and now wait for `1 - 0.5 = 0.5 seconds`. What do you think @kantp , am I making sense?

Reply:
@adinapoli-iohk afaik @Martoon-00 has done what you want. `concurrently` function waits when both actions are finished, so following your example: if payment takes `0.5` sec, the `newPayment` will work for 1 second because there is the second action `delay time` which isn't finished yet.

Reply:
Ah, gotcha, didn't appreciate fully the semantic of `concurrently` here, my bad!

Reply:
Yes, I think this is ok.

Reply:
Yes, looks exactly like what I had in mind.